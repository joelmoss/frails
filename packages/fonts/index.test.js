const serializer = require('jest-serializer-path')
const { createConfig } = require('@frails/core')

const fonts = require('.')
jest.mock('./require_resolve')

describe('fonts', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('createConfig', () => {
    expect(createConfig(fonts)).toMatchSnapshot()
    expect(JSON.stringify(createConfig(fonts()))).toBe(JSON.stringify(createConfig(fonts)))
  })

  test('disabled', () => {
    expect(createConfig(fonts(false))).toMatchSnapshot()
  })

  test('with options', () => {
    expect(createConfig(fonts({ name: '[path][name]-[hash].[ext]' }))).toMatchSnapshot()
  })
})
