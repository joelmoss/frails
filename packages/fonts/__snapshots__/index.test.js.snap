// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fonts createConfig 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Object {
          "loader": "file-loader",
          "options": Object {
            "mimetype": "application/font-woff",
          },
        },
      },
    ],
  },
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`fonts disabled 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`fonts with options 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Object {
          "loader": "file-loader",
          "options": Object {
            "mimetype": "application/font-woff",
            "name": "[path][name]-[hash].[ext]",
          },
        },
      },
    ],
  },
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;
